sub/mutation_events.go

```diff
@@ -0,0 +1,167 @@
+package sub
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/service/sqs"
+	"github.com/rs/zerolog"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/proto"
+)
+
+type EventType uint8
+
+const (
+	EventTypeCreated EventType = 1
+	EventTypeUpdated EventType = 2
+	EventTypeDeleted EventType = 3
+)
+
+// NewMutationEventSqsProcessor creates an SQS event processor that reads mutation events
+// from an SQS queue, Unmarshal them into protocol buffer messages, and processes them
+// using a strongly-typed event handler function.
+func NewMutationEventSqsProcessor[T proto.Message](
+	svc *sqs.Client,
+	queueURL string,
+	newMessage func() T,
+	handler EventMutationProtoHandlerFn[T],
+) *SqsEventProcessor {
+
+	snsString := MutationEventHandlerToStringHandler(handler, newMessage)
+	snsWrapper := StringHandlerToSnsWrapperHandler(snsString)
+
+	return &SqsEventProcessor{
+		svc:       svc,
+		queueURL:  queueURL,
+		logger:    zerolog.Nop(),
+		handlerFn: jsonEventHandlerToSqsHandlerFn(snsWrapper),
+	}
+}
+
+type EventMutationProtoHandlerFn[T proto.Message] func(context.Context, ProtoMutationEvent[T]) error
+
+// Represents a generic event with top level fields encoding occurring/generated in the system.
+type MutationEvent struct {
```

Comment:
Can you make this private so that people don't try to use it outside of the package.
```suggestion
type mutationEvent struct {
```